### DISCUSSION TITLE
What methods are used for avoiding detection?

### DISCUSSION THREAD

**Discussion by MortazaG** [ultrafunkamsterdam/undetected-chromedriver#1088] [General]
What methods does the undetected_chromedriver use for avoiding detection? I've been trying to find the answer but to no avail. Going through the code didn't make me any wiser.

When I go to a site like: https://httpbin.org/anything

All the data is the same every time I visit the site with undetected_chromedriver. I thought that it would change the user-agent everytime or something of the sort.

So how does it work in practice, in avoiding detection?

**Comment by damodharheadrun** [comment]
@MortazaG  Use the wait time / sleep time when we click the element or look for an element
